src/asmcup/genetics/Gene.java

```diff
@@ -10,16 +12,27 @@ public Gene(byte[] dna, int gen, float score) {
 		this.gen = gen;
 		this.score = score;
 	}
-	
+
+	@Override
 	public int compareTo(Gene other) {
 		float d = score - other.score;
 		
-		if (d == 0) {
+		if (Math.abs(d) < 0.00000001f) {

```

Comment:
> (Note: I'm not arguing because I'm trying to "fight" over this, just trying to make the intentions clear.)

Yeah, I know. I just thought this will get out of hand and you are probably right considering that I am not familiar with the Gene codebase. No hurt feelings!

> This cannot happen because the scores are final. 

This is what I was missing. :x 

---

The last paragraph clicked. Funny that I now finally understand as soon as I thought I'll stop spending my time on this particular issue :D.